test/test_prototype_builtin_datasets.py

```diff
@@ -75,6 +83,13 @@ def test_sample(self, test_home, dataset_mock, config):
         except Exception as error:
             raise AssertionError("Drawing a sample raised the error above.") from error
 
+        for dp in extract_datapipes(dataset):
+            if not isinstance(dp, _DemultiplexerIterDataPipe):
+                continue
+
+            for dq in dp.child_buffers:
+                dq.clear()
```

Comment:
This is similar to what @ejguan proposed in https://github.com/pytorch/vision/pull/5801#issuecomment-1098128946, but we still have [one failure](https://app.circleci.com/pipelines/github/pytorch/vision/16777/workflows/cf05d413-2eea-4a70-bf5a-8f8a6ec748ad/jobs/1359645).